### PR TITLE
vehr remmediation

### PR DESCRIPTION
**Pull request by omegasystemsgroup**
#### Short description of what this resolves:
This PR includes:
1. Removing external calls to 3rd Party servers
2. Includes audit logs
3. Remove initial dummy data

### CODE REVIEW

File: interface/modules/custom_modules/oe-module-visualehr/api/active_codes.php [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * Contains all of the Visual Dashboard global settings and configuration
+ *
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Kofi Appiah <kkappiah@medsov.com>
+ * @copyright Copyright (c) 2023 Visual EHR <https://visualehr.com/>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+$_SESSION['site_id'] = 'default';
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/active_codes.php:13]: Don't hard code this

File: interface/modules/custom_modules/oe-module-visualehr/api/active_codes.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+}
+$pid = preg_replace("#[^0-9]#", "", $_GET["pid"]);
+
+if ($method == 'GET') {
+    echo json_encode(getActiveIssues($pid));
+}
+
+function getColor()
+{
+    $colors = ["#eb5ea8", "#9f4ed3", "#99b1fc", "#7e83d0", "#fb4e4e", "#32a852", "#fb4e4e"];
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/active_codes.php:32]: These should be in a List, not hard coded

File: interface/modules/custom_modules/oe-module-visualehr/api/codetypes.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+ * Contains all of the Visual Dashboard global settings and configuration
+ *
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Kofi Appiah <kkappiah@medsov.com>
+ * @copyright Copyright (c) 2023 Visual EHR <https://visualehr.com/>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+$_SESSION['site_id'] = 'default';
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/codetypes.php:13]: Don't do this

File: interface/modules/custom_modules/oe-module-visualehr/api/color.php [hunk 4]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+function getColor()
+{
+    $colors = ["#eb5ea8", "#9f4ed3", "#99b1fc", "#7e83d0", "#fb4e4e", "#32a852", "#fb4e4e"];
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/color.php:5]: Better to put these in a List

File: interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+ * Contains all of the Visual Dashboard global settings and configuration
+ *
+ * @package openemr
+ * @link      http://www.open-emr.org
+ * @author    Kofi Appiah <kkappiah@medsov.com>
+ * @copyright Copyright (c) 2023 Visual EHR <https://visualehr.com/>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+$_SESSION['site_id'] = 'default';
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php:13]: No

File: interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+require_once "../../../../globals.php";
+
+
+use OpenEMR\Common\Logging\EventAuditLogger;
+
+
+$method = $_SERVER['REQUEST_METHOD'];
+
+//query occurence and assign it to the dropdown array()
+$sql = "SELECT option_id as id, list_id, title as name FROM `list_options` where list_id LIKE '%occurrence%' AND activity = 1";
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php:24]: Should use the ListService

File: interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    $_SESSION["authProvider"], //authProvider
+    $sql,
+    1,
+    'open-emr',
+    'dashboard'
+);
+
+
+//query outcome and assign it to the dropdown array()
+$sql = "SELECT option_id as id, list_id, title as name FROM `list_options` where list_id LIKE '%outcome%' AND activity = 1";
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php:42]: Use ListService

File: interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+    $_SESSION["authProvider"], //authProvider
+    $sql,
+    1,
+    'open-emr',
+    'dashboard'
+);
+
+
+//query verification_status and assign it to the dropdown array()
+$sql = "SELECT option_id as id, list_id, title as name FROM `list_options` where list_id LIKE '%condition-verification%' AND activity = 1";
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php:60]: ListService

File: interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    null, //pid
+    $_SESSION["authUser"], //authUser
+    $_SESSION["authProvider"], //authProvider
+    $sql,
+    1,
+    'open-emr',
+    'dashboard'
+);
+
+$sql = "SELECT option_id as id, list_id, title as name FROM `list_options` where list_id LIKE '%medication-usage-category%' AND activity = 1";
```
Comment (robertdown) [inline interface/modules/custom_modules/oe-module-visualehr/api/dropdown.php:76]: ListService

**Review by robertdown — CHANGES_REQUESTED** [review]
I started to review this PR but there are still so many things that prevent me from considering this that I had to stop.

Please review the ListService as it will reduce a lot of your code. I don't know why you are hard coding the site_id in every script, but by doing that you ensure multi-site setups do not work. There are hard coded values all over the place, put those in a List so the end user can customize things like the colors.

It seems you are attempting to recreate and API that already exists. For instance, problems.php has a function called getMedicalProblems - that functionality already exists in the core codebase, why are you recreating it? It appears that file is acting like a Controller, ensuring you can get json_encoded data out, which is fine, but don't recreate the already existing functions.

**Comment by robertdown** [comment]
Is there anymore planned work on this PR? If not, will close.

**Comment by omegasystemsgroup** [comment]
Please close. Thank you

**Pull request closed without merging**